src/neo4j_graphrag/embeddings/openai.py

```diff
@@ -53,10 +79,35 @@ def embed_query(self, text: str, **kwargs: Any) -> list[float]:
         response = self.openai_client.embeddings.create(
             input=text, model=self.model, **kwargs
         )
-        return response.data[0].embedding
+        embedding: list[float] = response.data[0].embedding
+        return embedding
 
 
-class AzureOpenAIEmbeddings(OpenAIEmbeddings):
-    def __init__(self, model: str = "text-embedding-ada-002", **kwargs: Any) -> None:
-        super().__init__(model, **kwargs)
-        self.openai_client = self.openai.AzureOpenAI(**kwargs)
+class AzureOpenAIEmbeddings(BaseOpenAIEmbeddings):
+    """
+    Azure OpenAI embeddings class.
+    This class uses the Azure OpenAI python client to generate embeddings for text data.
+
+    Args:
+        model (str): The name of the Azure OpenAI embedding model to use. Defaults to "text-embedding-ada-002".
+        kwargs: All other parameters will be passed to the openai.AzureOpenAI init.
+    """
+
+    def _initialize_client(self, **kwargs: Any) -> Any:
+        return self.openai.AzureOpenAI(**kwargs)
+
+    def embed_query(self, text: str, **kwargs: Any) -> list[float]:
```

Comment:
The embed query is exactly the same for both clients. You can put it directly in the Base class?

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/203/commits/4e86acbefbb530ac7bf61276a3b25ae6d402f7c2